### PR TITLE
libretranslate: init at 1.3.11

### PR DESCRIPTION
**Pull request by misuzu**
###### Description of changes
https://libretranslate.com/
https://github.com/LibreTranslate/LibreTranslate

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/libretranslate/default.nix [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+  # required for import check to work (argostranslate)
+  env = {
+    XDG_DATA_HOME = "/tmp";
+    XDG_CACHE_HOME = "/tmp";
+  };
+
+  pythonImportsCheck = [ "libretranslate" ];
+
+  meta = with lib; {
```
Comment (mweinelt) [inline pkgs/development/python-modules/libretranslate/default.nix:84]: Add a changelog pattern, please.

https://github.com/LibreTranslate/LibreTranslate/releases/tag/v1.3.11

File: pkgs/development/python-modules/libretranslate/default.nix [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  src = fetchFromGitHub {
+    owner = "LibreTranslate";
+    repo = "LibreTranslate";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-S2J7kcoZFHOjVm2mb3TblWf9/FzkxZEB3h27BCaPYgY=";
+  };
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/libretranslate/default.nix:42]: Move this down below `doCheck`.

Comment (misuzu) [reply pkgs/development/python-modules/libretranslate/default.nix]: Done!

File: pkgs/development/python-modules/libretranslate/default.nix (hunk 1 above)
Comment (misuzu) [reply pkgs/development/python-modules/libretranslate/default.nix:84]: Done!

File: pkgs/development/python-modules/argos-translate-files/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    ln -s */requires.txt requirements.txt
+  '';
+
+  # required for import check to work (argostranslate)
+  env = {
+    XDG_DATA_HOME = "/tmp";
+    XDG_CACHE_HOME = "/tmp";
+  };
+
+  pythonImportsCheck = [ "argostranslatefiles" ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/argos-translate-files/default.nix:34]:
```suggestion
  pythonImportsCheck = [ "argostranslatefiles" ];
  
  doCheck = false; # no tests
```

File: pkgs/development/python-modules/argostranslate/default.nix [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  '';
+
+  doCheck = false; # needs network access
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  # required for import check to work
+  # PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
```
Comment (mweinelt) [inline pkgs/development/python-modules/argostranslate/default.nix:50]: Likely means you need to fix up `HOME`.

File: pkgs/development/python-modules/libretranslate/default.nix [hunk 5]
```diff
@@ -0,0 +1,91 @@
+
+{ lib
```
Comment (mweinelt) [inline pkgs/development/python-modules/libretranslate/default.nix:2]:
```suggestion
{ lib
```

File: pkgs/development/python-modules/morfessor/default.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "bb3beac234341724c5f640f65803071f62373a50dba854d5a398567f9aefbab2";
+  };
+
+  pythonImportsCheck = [ "morfessor" ];
+
```
Comment (mweinelt) [inline pkgs/development/python-modules/morfessor/default.nix:18]: Set up an explicit test phase here.

> Morfessor> WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.

File: pkgs/development/python-modules/translatehtml/default.nix [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      --replace "==" ">="
+  '';
+
+  # required for import check to work (argostranslate)
+  env = {
+    XDG_DATA_HOME = "/tmp";
+    XDG_CACHE_HOME = "/tmp";
+  };
+
+  pythonImportsCheck = [ "translatehtml" ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/translatehtml/default.nix:37]: > translatehtml> Ran 0 tests in 0.000s

Disable check phase if there are no tests.

**Review by mweinelt — CHANGES_REQUESTED** [review]
Looks pretty good. Can we use SRI hashes on everything though?

**Comment by misuzu** [comment]
> Can we use SRI hashes on everything though?

I've took the hashes directly from PyPI pages

File: pkgs/development/python-modules/argostranslate/default.nix (hunk 4 above)
Comment (misuzu) [reply pkgs/development/python-modules/argostranslate/default.nix:50]: What do you mean? It's just creating a [bunch of directories on import](https://github.com/argosopentech/argos-translate/blob/401ca767db3cf3b6f2db5fb004488fa5b350defb/argostranslate/settings.py) in `~/.local`

Comment (mweinelt) [reply pkgs/development/python-modules/argostranslate/default.nix:50]: I mean the homeless-shelter error means, that it wants to access something below `$HOME`. And if that is required for the tests to work it should be set, e.g. to `HOME=$TMPDIR`.

Comment (misuzu) [reply pkgs/development/python-modules/argostranslate/default.nix:50]: Updated!

File: pkgs/development/python-modules/morfessor/default.nix (hunk 6 above)
Comment (misuzu) [reply pkgs/development/python-modules/morfessor/default.nix:20]: Done!

File: pkgs/development/python-modules/translatehtml/default.nix (hunk 7 above)
Comment (misuzu) [reply pkgs/development/python-modules/translatehtml/default.nix:34]: Done!

**Pull request merged**